src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/RedundantImportCheck.java

```diff
@@ -156,4 +189,39 @@ private static boolean isFromPackage(String importName, String pkg) {
         return front.equals(pkg);
     }
 
+    /**
+     * Determines if an import statement is a star import.
+     * @param imp the import
+     * @return whether imp is a start import
+     */
+    private static boolean isStarImport(FullIdent imp) {
+        return imp.getText().endsWith(".*");
+    }
+
+    /**
+     * Determines if a package imported by {@code imp} can be imported using {@code starImp}.
+     * @param starImp the star import
+     * @param imp the import
+     * @return whether package imported by {@code imp} can be imported using {@code starImp}
+     */
+    private static boolean isDuplicatedByStarImport(FullIdent starImp, FullIdent imp) {
+        final String importSplitRegexp = "\\.";
+
+        final String[] impSplitted = imp.getText().split(importSplitRegexp);
+        final String[] starImpSplited = starImp.getText().split(importSplitRegexp);
+
+        boolean isEquals = true;
+
+        if (impSplitted.length == starImpSplited.length) {
+            // check for common prefix
+            for (int i = 0; i < impSplitted.length - 1 && isEquals; i++) {
+                isEquals = impSplitted[i].equals(starImpSplited[i]);
```

Comment:
if  this code stays, you need to break the loop as soon as `isEquals` is `false`. If you don't you could have issues.
Example: Compare `a.b.c.d` to `a.e.c.d`.
Please try to find a case for this and add it as a test.

Reply:
> you need to break the loop as soon as isEquals is false

Actually, that's exactly what happens here. And the test case for it is https://github.com/checkstyle/checkstyle/pull/6461/files#diff-485a605456876d771b8df524551fe298
(we check that both imports have same amount of parts before this loop is executed and we don't check last part of both imports in the loop)

Reply:
Your right, didn't see the `&& isEquals` in the for loop. I was expecting it in the body.
This is resolved.